integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableDropColumnCommand.scala

```diff
@@ -55,6 +55,14 @@ private[sql] case class AlterTableDropColumnCommand(
       carbonTable = metastore
         .lookupRelation(Some(dbName), tableName)(sparkSession).asInstanceOf[CarbonRelation]
         .tableMeta.carbonTable
+      if (carbonTable.isPreAggregateTable) {
```

Comment:
we should check the children tables columns and throw the exception if drop column matches any child table column along with information of tables